### DISCUSSION TITLE
Is php 8.4 not yet supported?

### DISCUSSION THREAD

**Discussion by jdanino** [spatie/global-ray#55] [Q&A]
Global ray install does not work for php8.4. 

Seems like it's looking for a dump_8.4.phar file in ~/.composer/vendor/spatie/global-ray/dump-phars which does not exist...

Anything I can do on my end or is this something you need to address?

Thanks!

**Comment by freekmurze** [comment]
We now support PHP 8.4 👍

**Reply by jdanino** [reply]
The dump_8.4.phar file now exists and it's being included by the global loader, but when I call ray() I still get Attempted to call function "ray" from namespace ...

I've tried uninstalling and installing, restarting php etc.. Am I missing something?

**Reply by stevebauman** [reply]
@jdanino Where are you calling it from? Are you installing it into the correct php `.ini` file that is being registered with the PHP version you are calling `ray()` from?

I'm able to install the new version with PHP 8.4, but I get deprecation exceptions instead which I'm digging into:

<img width="968" alt="Screenshot 2024-12-11 at 11 56 32 AM" src="https://github.com/user-attachments/assets/52a54b67-1a69-429a-944d-45ab3d4c6d60" />

```

Deprecated: _HumbugBoxe4af63b3dc8f\Psy\info(): Implicitly marking parameter $config as nullable is deprecated, the explicit nullable type must be used instead in phar:///Applications/Tinkerwell.app/Contents/Resources/tinkerwell/tinker.phar/vendor/psy/psysh/src/functions.php on line 52

   _HumbugBoxe4af63b3dc8f\Whoops\Exception\ErrorException 

  voku\helper\ASCII::to_ascii(): Implicitly marking parameter $replace_single_chars_only as nullable is deprecated, the explicit nullable type must be used instead

  at vendor/voku/portable-ascii/src/voku/helper/ASCII.php:795
    791▕      *                <p>A string that contains only ASCII characters.</p>
    792▕      *
    793▕      * @phpstan-param ASCII::*_LANGUAGE_CODE $language
    794▕      */
  ➜ 795▕     public static function to_ascii(
    796▕         string $str,
    797▕         string $language = self::ENGLISH_LANGUAGE_CODE,
    798▕         bool $remove_unsupported_chars = true,
    799▕         bool $replace_extra_symbols = false,
```

**Comment by jdanino** [comment]
Just from a random controller. The auto_prepend with the global-loader shows up in phpinfo on the same page. The dump_8.4.phar is loaded but I still get the error above, and function_exists('ray') returns false 🤔
